### PR TITLE
Clean up and organize existing docs

### LINKED ISSUE #372: Document local dev env setup using Shared Infra tools
With the integration of Shared Infra tools and the `single-cell-infra` repo, new security practices should be followed and affect our development workflow. In order to continue development on private AWS resources, we will be using an ssh tunnel via Shared Infra's bastion server.

DoD:

- [ ] Update corpora-data-portal README with instructions to set up a local dev env using bastion

### PR DESCRIPTION
**Pull request by calvinnhieu**
Closes #372

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,70 +10,35 @@ such as [cellxgene](https://chanzuckerberg.github.io/cellxgene/posts/cellxgene_c
...
+
+### Development
+1. [Backend](backend/chalice/api_server/README.md#Development)
+1. [Frontend](frontend/README.md#Development)
+
+### Deployment
+1. Set `DEPLOYMENT_STAGE` and `AWS_PROFILE` according to the environment to be deployed.
 1. [Deploy Backend](backend/chalice/api_server/README.md#Deploy)
-1. [Deploy Cloudfront-invalidator](backend/chalice/cloudfront_invalidator/README.md#Deploy)
+1. [Deploy Cloudfront-invalidator](backend/chalice/api_server/README.md#Deploy)
```
Comment (maniarathi) [inline README.md:43]:
```suggestion
1. [Deploy Cloudfront-invalidator](backend/chalice/cloudfront_invalidator/README.md#Deploy)
```

File: backend/chalice/api_server/README.md [hunk 2]
```diff
@@ -27,9 +26,17 @@ Filename                  | Purpose                           | Information link
...
+     pip install -r requirements.txt
      ```
 
+1.  Enable local connection to the private `dev` RDS instance:
+
+     ```shell
+     ssh -L 5433:corpora-dev-corpora-api.cluster-c81u9imopfwl.us-west-2.rds.amazonaws.com:5432 bastion.dev.single-cell.czi.technology
+     ```
+
+    Note: This command creates an ssh tunnel from `localhost:5433` to the RDS connection endpoint via a `bastion` server.
```
Comment (maniarathi) [inline backend/chalice/api_server/README.md:38]: There was a whole shebang around this wasn't there when I tried to connect? I ended up needing to install blessclient?

File: docs/awscli.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    source_profile = czi-id
+    region = us-east-1
+
+    [profile single-cell-prod]
+    role_arn = arn:aws:iam::ACCOUNT_ID:role/poweruser
+    source_profile = czi-id
+    region = us-east-1
+    ```
+
+    Please contact #help-infra on Slack if you require access to `single-cell-dev` or `single-cell-prod`.
```
Comment (maniarathi) [inline docs/awscli.md:32]:
```suggestion
    Please contact [#help-infra](https://chanzuckerbergteam.slack.com/archives/C94RQ5SBV) on Slack if you require access to `single-cell-dev` or `single-cell-prod`.
```

File: docs/dev_flags.md [hunk 4]
```diff
@@ -0,0 +1,8 @@
+# Developer flags
+
+Set the following developer flags in your shell startup script of choice (e.g. `.bashrc`, `.bash_profile`, `.zshrc`):
+
+```bash
+# Enables local development of the Corpora app
+export CORPORA_LOCAL_DEV=true
+```
```
Comment (maniarathi) [inline docs/dev_flags.md:8]: Could also set the DEPLOYMENT_STAGE variable?

File: backend/chalice/api_server/README.md (hunk 2 above)
Comment (calvinnhieu) [reply backend/chalice/api_server/README.md:38]: Yeah, the first step in this section points to a "Pre-requisites" section that covers setting up `blessclient`. Sufficient?

File: docs/dev_flags.md (hunk 4 above)
Comment (calvinnhieu) [reply docs/dev_flags.md:8]: i.e. default to `dev`? Sure, though I felt that one was developer controlled whereas this one is meant to be constant. Specifically its purpose is differentiating a development deployment of the app from and production deployment. Right now, it's just being used to connect to the RDS via `localhost:5433` (development deployment) or the live RDS endpoint (production deployment).

Comment (maniarathi) [reply docs/dev_flags.md:8]: That's fair! Can disregard the suggestion :)

File: backend/chalice/api_server/README.md (hunk 2 above)
Comment (maniarathi) [reply backend/chalice/api_server/README.md:38]: Oh oops! I missed that; yep, that works, thanks!

**Pull request merged**